pandas/_libs/index.pyx

```diff
@@ -468,9 +467,8 @@ cdef class PeriodEngine(Int64Engine):
     cdef int64_t _unbox_scalar(self, scalar) except? -1:
         if scalar is NaT:
             return scalar.value
-        if isinstance(scalar, Period):
+        if isinstance(scalar, periodlib._Period):
```

Comment:
you would need to cimport _Period to get the performance improvement

Reply:
Just so I don't do anything stupid, I need to create the file ```period.pxd``` at ```pandas/_libs/tslibs/```, correct?

Reply:
yep